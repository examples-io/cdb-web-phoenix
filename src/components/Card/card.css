.card {
    display: grid;
}

.card .card-left {
    grid-column: 1 / span 1;
    margin-right: 2rem;
}

.card .card-right {
    grid-column: 2 / span 1;
}