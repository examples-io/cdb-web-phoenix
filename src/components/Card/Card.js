import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Image from '../Image/Image'
import './card.css'

/**
 * @description Card component - renders a div with "card" styling with specified text and optional image
 */
class Card extends Component {
    render() {
        return (
            <div className={'card ' + (this.props.className ? this.props.className : '')} id={this.props.id}>
                {
                    this.props.image &&
                    <div className="card-left">
                        <Image alt={this.props.image.alt} src={this.props.image.src} />
                    </div>
                }
                <div className="card-right">
                    {this.props.children}
                </div>
            </div>
        );
    }
}

Card.propTypes = {
    /**
     * children - text / nodes to be displayed on the card
     */
    chidren: PropTypes.node,
    /**
     * image - object with information about the image to be displayed.
     * Structure:
     *  {
     *      alt : string,
     *      src: string
     *  }
     */
    image: PropTypes.object,
    /**
     * id - element id of the card
     */
    id: PropTypes.string,
    /**
     * className - classes to be applied to the Card element
     */
    className: PropTypes.string
};

export default Card;