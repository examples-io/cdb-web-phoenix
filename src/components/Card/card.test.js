import React from 'react';
import renderer from 'react-test-renderer';

import Card from './Card';

describe('Card tests', () => {
    it ('renders Card without image correctly', () => {
        const tree = renderer
            .create(<Card id="testCard"> Test card </Card>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it ('renders Card with image correctly', () => {
        const tree = renderer
            .create(<Card id="testCardWithImage" image={{ alt: 'test card', src: '/test.png' }}> Test card with image </Card>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it ('renders Card with custom class correctly', () => {
        const tree = renderer
            .create(<Card id="testCardWithCustomClass" className="test-class"> Test card with image </Card>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});