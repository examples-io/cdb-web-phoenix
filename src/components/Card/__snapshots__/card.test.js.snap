// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests renders Card with custom class correctly 1`] = `
<div
  className="card test-class"
  id="testCardWithCustomClass"
>
  <div
    className="card-right"
  >
     Test card with image 
  </div>
</div>
`;

exports[`Card tests renders Card with image correctly 1`] = `
<div
  className="card "
  id="testCardWithImage"
>
  <div
    className="card-left"
  >
    <img
      alt="test card"
      className={undefined}
      id={undefined}
      src="/test.png"
    />
  </div>
  <div
    className="card-right"
  >
     Test card with image 
  </div>
</div>
`;

exports[`Card tests renders Card without image correctly 1`] = `
<div
  className="card "
  id="testCard"
>
  <div
    className="card-right"
  >
     Test card 
  </div>
</div>
`;
